sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyType.cs

```diff
@@ -3,60 +3,60 @@
 // license information.
 
 using System;
+using System.Collections.Generic;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
     /// Supported JsonWebKey key types (kty)
     /// </summary>
-    public enum KeyType : uint
+    public struct KeyType
     {
-        EllipticCurve = 0x0001,
-        EllipticCurveHsm = 0x0002,
-        Rsa = 0x0004,
-        RsaHsm = 0x0008,
-        Octet = 0x0010,
-        Other = 0x0020,
-    }
+        /// <summary>
+        /// Supported JsonWebKey key types (kty) as enum
+        /// </summary>
+        public enum KeyTypeValues : uint
```

Comment:
>public enum KeyTypeValues : uint [](start = 8, length = 32)

I would call this enum Simply Values since it's nested in the KeyType struct already.  This would make referencing it be KeyType.Values.Rsa vs. KeyType.KeyTypeValues.Rsa